tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsManager.cs

```diff
@@ -28,37 +27,49 @@ internal class DataStreamsManager
     private static readonly AsyncLocal<PathwayContext?> LastConsumePathway = new(); // saves the context on consume checkpointing only
     private readonly ConcurrentDictionary<string, RateLimiter> _schemaRateLimiters = new();
     private readonly NodeHashBase _nodeHashBase;
-    private bool _isEnabled;
     private IDataStreamsWriter? _writer;
+    private volatile DataStreamsState _state;
 
     public DataStreamsManager(
         string? env,
         string defaultServiceName,
-        IDataStreamsWriter? writer)
+        IDataStreamsWriter? writer,
+        DataStreamsState state)
     {
         // We don't yet support primary tag in .NET yet
         _nodeHashBase = HashHelper.CalculateNodeHashBase(defaultServiceName, env, primaryTag: null);
-        _isEnabled = writer is not null;
         _writer = writer;
+        _state = state;
     }
 
-    public bool IsEnabled => Volatile.Read(ref _isEnabled);
+    public enum DataStreamsState
+    {
+        Default = 0,
+        Enabled = 1,
+        Disabled = 2
+    }
```

Comment:
I wonder if it would make more sense to not expose this enum for use in consuming code. Instead, we could expose a set of features (as well as whether DSM _in general_ is Enabled). e.g. 

```csharp
public bool IsEnabled { get; }
public bool IsPayloadCalculationEnabled { get; }
```

That would remove all the confusion between "is disabled" vs "is not default" vs "is enabled", and what the implications of each of those are? 🤔